py/desitarget/lyazcat.py

```diff
@@ -0,0 +1,441 @@
+
+    
+def make_new_zcat(zbestname):
+    """Make the initial zcat array with redrock data.
+    
+    Parameters
+    ----------
+    zbestname : :class:`str`
+        Full filename and path for the zbest file to process.
+        
+    Returns
+    -------
+    :class:`~numpy.array`
+        A zcat in the official format (`zcatdatamodel`) compiled from
+        the `tile', 'night', and 'petal_num', in `zcatdir`.
+    """
+    tmark('    Making redrock zcat')
+    try:
+        zs = fits.open(zbestname)["ZBEST"].data
+        # EBL write out the zcat as a file with the correct data model.
+        zcat = np.zeros(len(zs), dtype={'names': zcatdatamodel_names, 'formats': zcatdatamodel_formats})
+        for col in zcols_copy:
+            zcat[col] = zs[col]
+        for col in n1_cols:
+            zcat[col][:] = -1
+
+        return zcat
+    except (FileNotFoundError, OSError):
+        return False
+
+    
+def add_qn_data(zcat, coaddname):
+    """Apply the QuasarNP model to the input zcat and add data to columns.
+    
+    Parameters
+    ----------
+    zcat : :class:'~numpy.array'
+        The structured array that was created by make_new_zcat()
+    coaddname : class:'str'
+        The name of the coadd file corresponding to the zbest file used
+        in make_new_zcat()
+        
+    Returns
+    -------
+    :class:'~numpy.array'
+        The zcat array with QuasarNP data included in the columns:
+        * Z_QN
+        * Z_QN_CONF
+        * IS_QSO_QN
+    """
+    tmark('    Adding QuasarNP data')    
+    lines = ['LYA','CIV(1548)','CIII(1909)', 'MgII(2796)','Hbeta','Halpha']
+    lines_bal = ['CIV(1548)']
+    model = load_model('/global/cfs/cdirs/desi/science/lya/qn_models/boss_dr12/qn_train_coadd_indtrain_0_0_boss10.h5')
+    
+    data, w = load_desi_coadd(coaddname)
+    data = data[:, :, None]
+    p = model.predict(data)
+    c_line, z_line, zbest, cbest, c_line_bal, z_line_bal = process_preds(p, lines, lines_bal)
+
+    c_thresh = 0.5
+    n_thresh = 1
+    is_qso = np.sum(c_line > c_thresh, axis=0) >= n_thresh
+
+    zcat['Z_QN'][w] = zbest
+    zcat['Z_QN_CONF'][w] = cbest
+    zcat['IS_QSO_QN'][w] = is_qso
+
+    return zcat
+
+
+def add_sq_data(zcat, coaddname, squeze_model):
+    """Apply the SQUEzE model to the input zcat and add data to columns.
+    
+    Parameters
+    ----------
+    zcat : :class:'~numpy.array'
+        The structured array that was created by make_new_zcat()
+    coaddname : class:'str'
+        The name of the coadd file corresponding to the zbest file used
+        in make_new_zcat()
+    squeze_model : :class:'numpy.array'
+        The loaded SQUEzE model file
+        
+    Returns
+    -------
+    :class:'~numpy.array'
+        The zcat array with SQUEzE data included in the columns:
+        * Z_SQ
+        * Z_SQ_CONF
+    """
+    tmark('    Adding SQUEzE data')
+    mdata = ["TARGETID"]
+    single_exposure = False
+    sq_cols_keep = ["PROB","Z_TRY","TARGETID"]
```

Comment:
Code style. Need a white-space after every comma in a list.

Reply:
Fixed for the next push.